sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -191,14 +196,18 @@ case class AdaptiveSparkPlanExec(
           case StageSuccess(stage, res) =>
             stage.resultOption = Some(res)
           case StageFailure(stage, ex) =>
-            errors.append(
-              new SparkException(s"Failed to materialize query stage: ${stage.treeString}." +
-                s" and the cause is ${ex.getMessage}", ex))
+            if (earlyFailedStage.contains(stage.id)) {
+              errors.prepend(
+                new SparkException(s"Early failed query stage found: ${stage.treeString}", ex))
```

Comment:
How about we handle it here? @maryannxue

Reply:
my whole point is...  we didn't need that `val earlyFailedStage` (i didn't mean the parameter in this method).

Reply:
I see, I see. Thanks:)